all-benchmarks/control/CFReT_benchmarks/memray_1.single_cell_processing.py

```diff
@@ -0,0 +1,150 @@
+                list(converted_dir.rglob(f"{name}_converted.parquet"))[0]
+            ).resolve(strict=True)
+        ),
+        "platemap_path": str(
+            pathlib.Path(list(platemap_dir.rglob(f"{name}_platemap.csv"))[0]).resolve(
+                strict=True
+            )
+        ),
+    }
+    for name in plate_names
+}
+
+# view the dictionary to assess that all info is added correctly
+pprint.pprint(plate_info_dictionary, indent=4)
+
+
+# In[4]:
+
+
+for plate, info in plate_info_dictionary.items():
+    print(f"Performing pycytominer pipeline for {plate}")
+    output_annotated_file = str(
+        pathlib.Path(f"{output_dir}/{plate}_sc_annotated.parquet")
+    )
+    output_normalized_file = str(
+        pathlib.Path(f"{output_dir}/{plate}_sc_normalized.parquet")
+    )
+    output_feature_select_file = str(
+        pathlib.Path(f"{output_dir}/{plate}_sc_feature_selected.parquet")
+    )
+
+    profile_df = pd.read_parquet(info["profile_path"])
+    platemap_df = pd.read_csv(info["platemap_path"])
+
+    # Step 1: Annotation
+    with memray.Tracker(f"{plate}_CFReT_annotate_benchmarks.bin", follow_fork=True):
+        annotate(
+            profiles=profile_df,
+            platemap=platemap_df,
+            join_on=["Metadata_well_position", "Image_Metadata_Well"],
+            output_file=output_annotated_file,
+            output_type="parquet",
+        )
+
+    # Load the annotated parquet file to fix metadata columns names
+    annotated_df = pd.read_parquet(output_annotated_file)
+
+    # Rename columns using the rename() function
+    column_name_mapping = {
+        "Image_Metadata_Site": "Metadata_Site",
+    }
+
+    annotated_df.rename(columns=column_name_mapping, inplace=True)
+
+    # Save the modified DataFrame back to the same location
+    annotated_df.to_parquet(output_annotated_file, index=False)
+
+    # set default for samples to use in normalization
+    samples = "all"
+
+    # Only for Plate 4, we want to normalize to the DMSO treatments
+    if plate == "localhost231120090001":
+        samples = "Metadata_heart_number == 7 and Metadata_treatment == 'DMSO'"
+
+    # Step 2: Normalization
+    with memray.Tracker(f"{plate}_CFReT_normalize_benchmarks.bin", follow_fork=True):
+        normalized_df = normalize(
+            profiles=output_annotated_file,
+            method="standardize",
+            output_file=output_normalized_file,
+            output_type="parquet",
+            samples=samples,
+        )
+
+    # Step 3: Feature selection
+    with memray.Tracker(
+        f"{plate}_CFReT_feature_select_benchmarks.bin", follow_fork=True
+    ):
+        feature_select(
+            output_normalized_file,
+            operation=feature_select_ops,
+            output_file=output_feature_select_file,
+            output_type="parquet",
+        )
+    print(
+        f"Annotation, normalization, and feature selection have been performed for {plate}"
+    )
+
+
+# In[5]:
+
+
+# Check output file
```

Comment:
Are there any assertions or tests that can be used here to check the file further?

Reply:
This is script is not written by me. These scripts are within the CFReT repo. The only thing I have added is the memory tracking with `memray` with the `Tracker` context manager. 

You can find the script [here](https://github.com/WayScience/CFReT_data/blob/main/3.process_cfret_features/scripts/1.single_cell_processing.py)

The only reason I have added these scripts is if other users want to conduct the benchmarks themselves, which is explained within the README of the repo.